drivers/serial/uart_nrfx_uart.c

```diff
@@ -215,6 +238,18 @@ static int uart_nrfx_fifo_read(struct device *dev,
 static void uart_nrfx_irq_tx_enable(struct device *dev)
 {
 	nrf_uart_int_enable(NRF_UART0, NRF_UART_INT_MASK_TXDRDY);
+
+	/* Critical section is used to avoid any UART related interrupt which
+	 * can occure after the if statement and before call of the function
+	 * forcing an interrupt. */
+	irq_lock();
```

Comment:
You need to store the key you get with `irq_lock()` and provide it to `irq_unlock()` later

Reply:
fixed